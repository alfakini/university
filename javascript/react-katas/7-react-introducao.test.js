import React from 'react';
import renderer from 'react-test-renderer';

// Implemente o componente React que renderiza uma receita de Tofu Assado
// Na pasta __snapshots__ você vai encontrar o componente alvo renderizado

describe('React', function() {
  const items = [
    {
      nome: "Tofu Assado",
      ingredientes: [
        "1 Tofu",
        "1 Copo de Noses",
        "2 Copos Alface",
        "1 Abóbora Amarela",
        "1/2 Copo de Óleo de Oliva",
        "3 Dentes de Alho"
      ],
      preparo: [
        "Pré-aqueça o forno a 350 graus.",
        "Espalhe o óleo de oliva em torno de uma vasilha de vidro.",
        "Adicione o Tofu, o alho e as nooses ao prato.",
        "Asse por 15 minutos.",
        "Adicione a abóbora amarela e coloque-a no forno por 30 minutos.",
        "Retire do forno e deixe esfriar por 15 minutos. Adicione a alface e sirva."
      ]
    },
    {
      nome: "Chocolate quente",
      ingredientes: [
        "2 colheres de sopa de abóbora (cozinhada a vapor)",
        "1/2 colher de chá de mistura de especiarias (cardamomo, canela, noz moscada, pimenta preta)",
        "1 tâmara medjool",
        "2 colheres de sopa de essência de baunilha ou vagem de baunilha",
        "2 +1/2 chávenas de bebida vegetal de avelã",
        "1 colher de cacau em pó",
        "Chantili vegan para servir"
      ],
      preparo: [
        "Num tacho pequeno cozinhar a abóbora com as especiarias, cerca de dois minutos mexendo sempre.",
        "Picar a tâmara e adicionar à abóbora.",
        "Juntar a baunilha em vagem ou essência e continuar a mexer.",
        "Adicionar a bebida vegetal à abóbora e continuar a mexer, sem deixar aquecer.",
        "Colocar a mistura num liquidificador e bater até ficar bem liquido. (Se utilizares vagem de baunilha, retira antes de liquidificar)",
        "Aquecer 1/2 chávena de bebida vegetal e adicionar o chocolate",
        "Juntar com a abóbora e mexer",
        "Decorar com chantili ou marshmallows vegan."
      ]
    }
  ]

  function createReceitas(receitas) {
    return // IMPLEMENTE. Implemente qualquer outra função extra que achar necessário
  }

  it('renderiza corretamente o componente gerado pela função createReceitas', function() {
    const component = renderer.create(createReceitas(items));
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
})
