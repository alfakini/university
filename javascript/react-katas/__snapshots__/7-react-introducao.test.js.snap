// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`React renderiza corretamente o componente gerado pela função createReceitas 1`] = `
<section>
  <h1>
    Receitas
  </h1>
  <section>
    <h1>
      Tofu Assado
    </h1>
    <h2>
      Ingredientes
    </h2>
    <ul
      className="ingredientes"
    >
      <li>
        1 Tofu
      </li>
      <li>
        1 Copo de Noses
      </li>
      <li>
        2 Copos Alface
      </li>
      <li>
        1 Abóbora Amarela
      </li>
      <li>
        1/2 Copo de Óleo de Oliva
      </li>
      <li>
        3 Dentes de Alho
      </li>
    </ul>
    <h2>
      Preparo
    </h2>
    <ul
      className="preparo"
    >
      <li>
        Pré-aqueça o forno a 350 graus.
      </li>
      <li>
        Espalhe o óleo de oliva em torno de uma vasilha de vidro.
      </li>
      <li>
        Adicione o Tofu, o alho e as nooses ao prato.
      </li>
      <li>
        Asse por 15 minutos.
      </li>
      <li>
        Adicione a abóbora amarela e coloque-a no forno por 30 minutos.
      </li>
      <li>
        Retire do forno e deixe esfriar por 15 minutos. Adicione a alface e sirva.
      </li>
    </ul>
  </section>
  <section>
    <h1>
      Chocolate quente
    </h1>
    <h2>
      Ingredientes
    </h2>
    <ul
      className="ingredientes"
    >
      <li>
        2 colheres de sopa de abóbora (cozinhada a vapor)
      </li>
      <li>
        1/2 colher de chá de mistura de especiarias (cardamomo, canela, noz moscada, pimenta preta)
      </li>
      <li>
        1 tâmara medjool
      </li>
      <li>
        2 colheres de sopa de essência de baunilha ou vagem de baunilha
      </li>
      <li>
        2 +1/2 chávenas de bebida vegetal de avelã
      </li>
      <li>
        1 colher de cacau em pó
      </li>
      <li>
        Chantili vegan para servir
      </li>
    </ul>
    <h2>
      Preparo
    </h2>
    <ul
      className="preparo"
    >
      <li>
        Num tacho pequeno cozinhar a abóbora com as especiarias, cerca de dois minutos mexendo sempre.
      </li>
      <li>
        Picar a tâmara e adicionar à abóbora.
      </li>
      <li>
        Juntar a baunilha em vagem ou essência e continuar a mexer.
      </li>
      <li>
        Adicionar a bebida vegetal à abóbora e continuar a mexer, sem deixar aquecer.
      </li>
      <li>
        Colocar a mistura num liquidificador e bater até ficar bem liquido. (Se utilizares vagem de baunilha, retira antes de liquidificar)
      </li>
      <li>
        Aquecer 1/2 chávena de bebida vegetal e adicionar o chocolate
      </li>
      <li>
        Juntar com a abóbora e mexer
      </li>
      <li>
        Decorar com chantili ou marshmallows vegan.
      </li>
    </ul>
  </section>
</section>
`;
